utilities/trace_analyzer_tool.cc

```diff
@@ -0,0 +1,1751 @@
+DEFINE_bool(output_qps_stats, false,
+            "Output the query per second(qps) statistics"
+            "For the overall qps, it will contain all qps of each query type.\n"
+            "For each cf and query, it will have its own qps output\n"
+            "format:[query_count_in_this_second].");
+DEFINE_bool(no_print, false, "Do not print out any result");
+DEFINE_string(
+    output_correlation, "",
+    "intput format: [correlation pairs][.,.]\n"
+    "Output the query correlations between the pairs of query types"
+    "listed in the parameter, input should select the operations from:\n"
+    "get, put, delete, single_delete, rangle_delete, merge. No space\n"
+    "between the pairs separated by commar. Example: =[get,get]...");
+DEFINE_string(key_space_dir, "",
+              "<the directory stores full key space files>"
+              "The key space files should be: <column family id>.txt");
+DEFINE_bool(analyze_get, false, "Analyze the Get query.");
+DEFINE_bool(analyze_put, false, "Analyze the Put query.");
+DEFINE_bool(analyze_delete, false, "Analyze the Delete query.");
+DEFINE_bool(analyze_single_delete, false, "Analyze the SingleDelete query.");
+DEFINE_bool(analyze_range_delete, false, "Analyze the DeleteRange query.");
+DEFINE_bool(analyze_merge, false, "Analyze the Merge query.");
+DEFINE_bool(analyze_iterator, false,
+            "Analyze the iterate query like seek() and seekForPre().");
+DEFINE_bool(no_key, true,
+            " Does not output the key to the result files to make smaller.");
+DEFINE_bool(print_overall_stats, true,
+            " Print the stats of the whole trace,"
+            "like total requests, keys, and etc.");
+DEFINE_bool(print_key_distribution, false, "Print the key size distribution.");
+DEFINE_bool(output_value_distribution, false,
+            "Print the value size distribution, only available for Put.");
+DEFINE_int32(print_top_k_access, 1,
+             "<top K of the variables to be printed>"
+             "Print the top k accessed keys, top k accessed prefix"
+             "and etc.");
+DEFINE_int32(output_ignore_count, 0,
+             "<threshold>, ignores the access count <= this value,"
+             "it will shorter the output.");
+DEFINE_int32(value_interval, 8,
+             "To output the value distribution, we need to set the value"
+             "intervals andmake the statistic of the value size distribution"
+             "in different intervals. The default is 8.");
+
+namespace rocksdb {
+
+std::map<std::string, int> taOptToIndex = {
+    {"get", 0},          {"put", 1},   {"delete", 2},  {"single_delete", 3},
+    {"range_delete", 4}, {"merge", 5}, {"iterator", 6}};
+
+std::map<int, std::string> taIndexToOpt = {
+    {0, "get"},          {1, "put"},   {2, "delete"},  {3, "single_delete"},
+    {4, "range_delete"}, {5, "merge"}, {6, "iterator"}};
+
+// Transfer the Microsecond time to date time
+std::string TraceAnalyzer::MicrosdToDate(uint64_t time_in) {
```

Comment:
We have existing code printing out date+time: https://github.com/facebook/rocksdb/blob/master/env/posix_logger.h#L106-L120

please consider to reuse the existing code.